src/gpdma.rs

```diff
@@ -0,0 +1,550 @@
+//! The GPDMA is the general purpose DMA engine in use on the STM32H5 family of processors. It is
+//! used to perform programmable data transfers that are offloaded from the CPU to the DMA engine.
+//!
+//! The GPDMA can perform the following transfers from a *source* address to a *destination*
+//! address:
+//! - Memory to memory
+//! - Memory to peripheral
+//! - Peripheral to memory
+//! - Peripheral to peripheral
+//!
+//! Each GPDMA has 8 channels. Each channel can service any hardware request (or memory to memory
+//! transfer) that is supported by the processor (ie. they're not tied to specific channels). All
+//! channels support direct and linked-buffer transfers. However, the channels do have different
+//! capabilities (see RM0492 Rev 3 section 15.3.2 for full details), notably that channels 0-5 can
+//! only service transfers in a linear address space, while channels 6 & 7 can service transfers
```

Comment:
Perhaps a clarification?

```suggestion
//! only service transfers in a linear address space, while channels 6 & 7 can also service transfers
```